docs/configuration.md

```diff
@@ -1 +1,78 @@
+
+### Conventions
+The spec of the configuration file must use these conventions.
+
+- Use **`[]`** to indicate an empty list
+- Use **`null`** to indicate a null value
+- Use **`true`** and **`false`** as only options on boolean fields
+
+
+## Data Sources
+
+Followings are dozer supported connectors and detailed example configurations are available:
+- [Postgres](/docs/configuration/sources/postgres#configuration)
+- [Snowflake](/docs/configuration/sources/snowflake#configuration)
+- [Ethereum](/docs/configuration/sources/ethereum#configuration)
+- [Kafka](/docs/configuration/sources/kafka#configuration)
+- [gRPC](/docs/configuration/sources/grpc#configuration)
+- [Delta Lake](/docs/configuration/sources/object-stores/formats/deltalake#configuration)
+- [AWS S3](/docs/configuration/sources/object-stores/types/aws-s3#configuration)
+- Local Object Store ([csv](/docs/configuration/sources/object-stores/formats/csv#configuration), [parquet](/docs/configuration/sources/object-stores/formats/parquet#configuration))
+
+And gather the necessary connection details for each data source. These details typically include:
+- Database hostname or endpoint
+- Port number
+- Database name
+- Username and password
+- Authentication method (if applicable)
+- Data warehouse or storage service credentials (if applicable)
+- API keys or tokens (if applicable)
+
+## SQL Transformations
+
+Define the data transformation rules you want to apply to the ingested data. 
+These SQLs might involve joining tables, aggregating data, or applying filters.
+
+## API Endpoints
+
+Configure the Store/API nodes to expose gRPC and REST APIs for data access. Define the endpoints and routes that users or downstream systems will use to query and retrieve data.
+
+## Global settings (Optional)
+
+| Property                               | Details                                                      | Type    | Default Value | Example                              |
+|----------------------------------------|--------------------------------------------------------------|---------|---------------|--------------------------------------|
+| `app_buffer_size`                      | Pipeline buffer size                                         | Integer | 20000         | `app_buffer_size: 50000`             |
+| `commit_size`                          |                                                              | Integer | 10000         | `commit_size: 100`                   |
```

Comment:
We should explain what is it

Reply:
I will fill up the missing description by getting from the person who implemented it - give me sometime!